File: go/1047E-删除字符串中的所有相邻重复项.go
/*
 * @Author: SukiEva
 * @Date: 2021-10-23 16:10:59
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func removeDuplicates(s string) string {
	stack := make([]byte, 0)
	for _, v := range s {
		if len(stack) != 0 && byte(v) == stack[len(stack)-1] {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, byte(v))
		}
	}
	return string(stack)
}

File: cpp/1047E-删除字符串中的所有相邻重复项.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-18 09:26:38
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    string removeDuplicates(string s) {
        string st;
        for (char c : s) {
            if (!st.empty() && st.back() == c)
                st.pop_back();
            else
                st.push_back(c);
        }
        return st;
    }
};